src/node.jl

```diff
@@ -1,6 +1,12 @@
 
 Base.@kwdef mutable struct NodeOptions
     rate::Float64 = 100
+    default_addr::Sockets.IPv4 = ip"127.0.0.1"
+    heartbeat_enable::Bool = false
+    heartbeat_addr::Sockets.IPv4 = default_addr
+    heartbeat_port::Int = getdefaultport()
+    heartbeat_rate::Float64 = 1.0  # Hz
+    heartbeat_print_rate_enable::Bool = false
```

Comment:
It feels a bit weird for these heartbeat options to be inside of `NodeOptions` rather than simply a part of `NodeHeartBeat`. Why not make `NodeHeartBeat` inherit from `Publisher` and contain all of these options directly in `NodeHeartBeat`.